sdk_extensions/tracing_incubator/src/main/java/io/opentelemetry/sdk/extensions/incubator/trace/data/SpanData.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.extensions.incubator.trace.data;
+
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.common.ReadableAttributes;
+import io.opentelemetry.sdk.common.InstrumentationLibraryInfo;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.trace.Span.Kind;
+import io.opentelemetry.trace.Status;
+import io.opentelemetry.trace.TraceFlags;
+import io.opentelemetry.trace.TraceState;
+import java.util.List;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * A {@link io.opentelemetry.sdk.trace.data.SpanData} implementation with a builder that can be used
+ * to modify parts of a {@link io.opentelemetry.sdk.trace.data.SpanData}.
+ *
+ * <pre>{@code
+ * String clientType = ClientConfig.parseUserAgent(
+ *   data.getAttributes().get(SemanticAttributes.HTTP_USER_AGENT).getStringValue());
+ * Attributes newAttributes = Attributes.newBuilder(data.getAttributes())
+ *   .setAttribute("client_type", clientType)
+ *   .build();
+ * data = io.opentelemetry.sdk.extensions.incubator.trace.data.SpanData.newBuilder(data)
+ *   .setAttributes(newAttributes)
+ *   .build();
+ * exporter.export(data);
+ *
+ * }</pre>
+ */
+// AutoValue generated hashCode is fine but we need to define equals to accept the base SpanData
+// type.
+@Immutable
+@AutoValue
+public abstract class SpanData implements io.opentelemetry.sdk.trace.data.SpanData {
```

Comment:
Can we name this something else, so we don't have the name collision, requiring using FQNs for hte interface?

Reply:
Not sure why I missed this review, sorry! I could have sworn I added a comment in description about it but didn't - I agree, but couldn't come up with a better name :( Any ideas?

Reply:
BuildableSpanData? SpanDataExtender? BuiltSpanData? 

Maybe just SpanDataBuilder for now, since it's really a temporary move to see if this is the best approach?